modules/snooker/src/main/ts/ephox/snooker/api/Warehouse.ts

```diff
@@ -31,13 +32,34 @@ const filterItems = function (warehouse: Warehouse, predicate: (x: Structs.Detai
   return Arr.filter(all, predicate);
 };
 
+const generateColumns = <T extends Structs.Detail> (rowData: Structs.RowData<T>, columnsGroup: Record<string, Structs.Column>) => {
+  const columns = Traverse.children(rowData.element);
+  const filteredColumns = Arr.filter(columns, SugarNode.isTag('col'));
+
+  let index = 0;
+
+  Arr.each(filteredColumns, (column: SugarElement<HTMLTableColElement>): void => {
+    const colspan = column.dom.getAttribute('colspan');
```

Comment:
Can you please use the appropriate sugar function here instead? eg `Attribute.get(column, 'colspan')`

Reply:
This also could maybe use `CellUtils.getSpan` instead to remove more duplication.